### PR TITLE
Product page

### PR DESCRIPTION
**Pull request by Shubhamkr2610**
Filtered Product 
I have filtered the products by categories and price  . So ignore other files.

Files to check
src/components/sidebar/sidebar.jsx
src/filterFunction/categoryFilter.js
src/pages/product-page/productPage.js
src/filterFunction/sortFilter.js
src/reducer/filterReducer.js

### CODE REVIEW

File: src/context/ProductContext.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    useEffect(()=>{
+      fetch('api/categories')
+      .then(res=>res.json())
+      .then(data=>setCategory(data.categories))
+    }, [])
+  
+
+    return(
+      <ProductContext.Provider value={{product: product , category: category}} >
```
Comment (Muzammil-khan710) [inline src/context/ProductContext.js:24]: You can use value={{product, category}}
It willl work

**Review by Muzammil-khan710 — APPROVED** [review]
Functionality of the filters are upto the mark !
Good code structure
Good to merge

**Review by Rockyraven — APPROVED** [review]
All goods to me
you are good to go

File: src/context/ProductContext.js (hunk 1 above)
Comment (Shubhamkr2610) [reply src/context/ProductContext.js:24]: yes ,, I can . nice pick . will fix it

File: src/components/product-card/ProductCard.css [hunk 2]
```diff
@@ -19,4 +19,10 @@
     margin: 0rem 0rem 0rem 1rem;
     justify-content: space-between;
     justify-items: center;
+}
+.product-rating{
+    border: 1px solid ;
+    border-radius: 10%;
+    background-color: var( --ecomm-color-border);
```
Comment (arjun-computer-geek) [inline src/components/product-card/ProductCard.css:26]: Nicely used colour variables

File: src/App.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ const App = () => {
    <>
     <Header/>     
       <Routes>
-         <Route path="/" element={<LandingPage/>} />
+         <Route path="/" element={<LandingPage/>} exact />
```
Comment (arjun-computer-geek) [inline src/App.js:11]: You can also add * path and 404 not found page

File: src/context/ProductContext.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+const ProductContext = createContext()
+
+
+export const ProductProvider = ({children}) => {
+      
+    const[ product , setProduct] = useState([])
+    const[category , setCategory] = useState([])
+    useEffect(()=>{
+        fetch ('/api/products')
```
Comment (arjun-computer-geek) [inline src/context/ProductContext.js:11]: You can also use axios for that it is easy.

File: src/components/sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import React from 'react'
 import './sidebar.css'
 
-export const Sidebar = () => {
+export const Sidebar = ({ dispatch, state}) => {
+    console.log(state.checked)
```
Comment (arjun-computer-geek) [inline src/components/sidebar/Sidebar.jsx:5]: Wahy are you passing  state and dispatch as a props? Is there need of this?

File: src/filterFunction/categoryFilter.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+export const categoryFiltered = (product,{Sports , Casual , Formal}) =>{
+    let categoryFilteredProduct = []; 
+
+    if (Sports === true)
```
Comment (arjun-computer-geek) [inline src/filterFunction/categoryFilter.js:4]: You can directly put here sports if it is holding boolean value.

Like:
If(Sports){ \\Do your stuff}

No need to do : sports === true

**Review by arjun-computer-geek — APPROVED** [review]
🔥🔥🔥🔥You have done great job @Shubhamkr2610 🔥🔥🔥🔥

There is some comments. Please consider it. Else other are awesome.

File: src/App.js (hunk 3 above)
Comment (Shubhamkr2610) [reply src/App.js:11]: will add it . Thank you

File: src/context/ProductContext.js (hunk 4 above)
Comment (Shubhamkr2610) [reply src/context/ProductContext.js:11]: will change it for time being ,  running late for now , have sort of time

File: src/filterFunction/categoryFilter.js (hunk 6 above)
Comment (Shubhamkr2610) [reply src/filterFunction/categoryFilter.js:4]: i have not idea about this . thanks for explanation , will try it

**Pull request merged**